Revit_Core_Engine/Query/DoesIntersect.cs

```diff
@@ -0,0 +1,176 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+
+using Autodesk.Revit.DB;
+using BH.oM.Base.Attributes;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Check if two elements intersect.")]
+        [Input("element1", "First element to check the intersection for.")]
+        [Input("element2", "Second element to check the intersection for.")]
+        [Output("bool", "Result of the intersect checking.")]
+        public static bool DoesIntersect(this Element element1, Element element2)
+        {
```

Comment:
It tempts me to day that this method should be as simple as this (for all cases link vs host etc.), more or less (code typed by hand):
```C#
        Transform transform1 = element1.Document.LinkTransform(); //Null if doc not linked
        Transform transform2 = element2.Document.LinkTransform();
        List<Solid> solids = element1.Solids();
        if (transform1 != null)
            solids = solids.Transform(transform1); //There is no such a method, but ykwim

        if (transform2 != null)
            solids = solids.Transform(transform2.Inverse);

        //Possibly some more pre-checks with bounding boxes here to improve performance
        foreach (Solid solid in solids)
        {
            ElementIntersectsSolidFilter intersectFilter = new ElementIntersectsSolidFilter(solid);
            if (intersectFilter.PassesFilter(element2))
                return true;
        }

I may be wrong but it sounds as if it could be generalised a lot
```